src/pseudopeople/interface.py

```diff
@@ -56,51 +63,98 @@ def _generate_dataset(
         source = Path(source)
         validate_source_compatibility(source)
 
-    data_paths = fetch_filepaths(dataset, source)
-    if not data_paths:
-        raise DataSourceError(
-            f"No datasets found at directory {str(source)}. "
-            "Please provide the path to the unmodified root data directory."
-        )
-    validate_data_path_suffix(data_paths)
-    noised_dataset = []
-    iterator = (
-        tqdm(data_paths, desc="Noising data", leave=False)
-        if len(data_paths) > 1
-        else data_paths
-    )
+    engine = get_engine_from_string(engine)
+
+    if engine == PANDAS:
+        # We process shards serially
+        data_paths = fetch_filepaths(dataset, source)
+        if not data_paths:
+            raise DataSourceError(
+                f"No datasets found at directory {str(source)}. "
+                "Please provide the path to the unmodified root data directory."
+            )
 
-    for data_path_index, data_path in enumerate(iterator):
-        logger.debug(f"Loading data from {data_path}.")
-        data = _load_data_from_path(data_path, user_filters)
-        if data.empty:
-            continue
-        data = _reformat_dates_for_noising(data, dataset)
-        data = _coerce_dtypes(data, dataset)
-        # Use a different seed for each data file/shard, otherwise the randomness will duplicate
-        # and the Nth row in each shard will get the same noise
-        data_path_seed = f"{seed}_{data_path_index}"
-        noised_data = noise_dataset(dataset, data, configuration_tree, data_path_seed)
-        noised_data = _extract_columns(dataset.columns, noised_data)
-        noised_dataset.append(noised_data)
-
-    # Check if all shards for the dataset are empty
-    if len(noised_dataset) == 0:
-        raise ValueError(
-            "Invalid value provided for 'state' or 'year'. No data found with "
-            f"the user provided 'state' or 'year' filters at {data_path}."
+        validate_data_path_suffix(data_paths)
+
+        # Iterate sequentially
+        noised_dataset = []
+        iterator = (
+            tqdm(data_paths, desc="Noising data", leave=False)
+            if len(data_paths) > 1
+            else data_paths
         )
-    noised_dataset = pd.concat(noised_dataset, ignore_index=True)
 
-    # Known pandas bug: pd.concat does not preserve category dtypes so we coerce
-    # again after concat (https://github.com/pandas-dev/pandas/issues/51362)
-    noised_dataset = _coerce_dtypes(noised_dataset, dataset, cleanse_int_cols=True)
+        for data_path_index, data_path in enumerate(iterator):
+            logger.debug(f"Loading data from {data_path}.")
+            data = load_standard_dataset(data_path, user_filters, engine=engine, is_file=True)
```

Comment:
How is the `is_file` flag different from just `engine == "PANDAS"`?

Reply:
They are always the same in our current usage, but need not be in the future, since they are conceptually different. Right now, when we run with Pandas, we load files one at a time, but Pandas (like Modin) can also load Parquet directories.

I have [asked in the past](https://ihme.slack.com/archives/C02KUQ9LX32/p1693499527569429) why we don't load the data all at once with Pandas -- still don't really understand this.

Reply:
If this is in fact providing a speed-up feel free to create a ticket for that work, ideally with details about that